fahrschein/src/test/java/org/zalando/fahrschein/NakadiReaderTest.java

```diff
@@ -536,7 +543,66 @@ public void shouldSendCursorsForLockedPartitions() throws IOException {
 
         nakadiReader.run();
 
+        // todo: This assertion is never reached.
         assertEquals(singletonList("[{\"partition\":\"0\",\"offset\":\"0\"},{\"partition\":\"1\",\"offset\":\"10\"}"), headers.get("X-Nakadi-Cursors"));
     }
 
+    @Test
+    public void shouldCallMetricsCollectorWithoutMetadataBeingPresent() throws Exception {
+        final ClientHttpResponse response = mock(ClientHttpResponse.class);
```

Comment:
I was thinking of extracting some of the common setup that gets repeated in almost every test to a setup method. How do you feel about it?